### PR TITLE
v061 add  "file-values.json"

### PR DESCRIPTION
**Pull request by pholleway**
**Related Issue(s):** v061 add "file-values.json"

**Description:** update all categorical classification fields 

**PR checklist:**

- [ ] Code is formatted (run `scripts/format`).
- [ ] Code lints properly (run `scripts/lint`).
- [ ] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [ ] Example STAC Catalog has been updated to reflect changes, if applicable.
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

**Comment by pholleway** [comment]
The most recent commit is on MXD14- Fire Product.

**Review by gadomski — CHANGES_REQUESTED** [review]
So I think we'll have to add `file:values` to a new fragment, rather than item properties, because the `file:values` apply to an asset and not an item: https://github.com/stac-extensions/file#mapping-object.

**Review by gadomski — COMMENTED** [review]
Awesome, this all looks great. Do you think there's any other products that we'll need to add file values for? Or are the fire ones our only victims?

I'll wait for full approval until we get integrate these into `stactools.modis.stac.create_item`.

**Comment by pholleway** [comment]
> Awesome, this all looks great. Do you think there's any other products that we'll need to add file values for? Or are the fire ones our only victims?
> 
> I'll wait for full approval until we get integrate these into `stactools.modis.stac.create_item`.

Sounds good, Pete! There are several more. I've committed the MCD product. MOD are done, but I want to review them. They should be committed soon. Then there's the MYD products to wrap up.

**Review by gadomski — CHANGES_REQUESTED** [review]
So turns out we're not going to use `file:values` for the `Fpar` psuedo-fill values, so can you empty the fragment files for:
- MCD16A2H
- MCD15A3H

Thanks!

**Comment by pholleway** [comment]
> So turns out we're not going to use `file:values` for the `Fpar` psuedo-fill values, so can you empty the fragment files for:
> 
> * MCD16A2H
> * MCD15A3H
> 
> Thanks!

Emptied the "file-values.json" fragment files  MCD15A2H and MCD15A3H. I believe the same issues will the continuous data applies to most datasets. There will be several files emptied.

**Comment by pholleway** [comment]
I'm curious if you had the chance to review MCD43A4 and MCD64A1. On MCD43A4 does have some discrete values (0 and 1), but there's also the 255 fill value. @gadomski

**Comment by gadomski** [comment]
> I'm curious if you had the chance to review MCD43A4 and MCD64A1. On MCD43A4 does have some discrete values (0 and 1), but there's also the 255 fill value.

I think because they're all discrete (no continuous values) that they're ok to represent this way.

**Comment by pholleway** [comment]
I believe the MCD15A2H and MCD15A3H 255-248/49 applies to classification. And not the actual percent 0-100 data. 

"file-values.json" left blank for these datasets until we figure out how to deal with continuous data.

**Comment by pholleway** [comment]
There are now "file-info.json" (actual discrete data) and "classification.json" (discrete categorical describing actual data) files for all MCD, MOD, and MYD datasets. The files were only filled out for existing discrete values (some files with either continuous or no data are empty). 

The tech docs on this metadata lumps actual data, classification, and fillvalues (nodata) together. This added to the struggle of sorting through each. 

Do we want to include fillvalues (nodata) at all? Maybe within in the "classification.json" or a dedicated .json file? Or just skip it for now?

The populated "file-info.json" for actual discrete data are listed below. 

- MCD43A4 (BRDF_Albedo_Band_Mandatory_Quality_Band1-7)

- MOD10A1 (NDSI_Snow_Cover_Basic_QA)
- MOD10A2(Maximum_Snow_Extent)(Eight_Day_Snow_Cover)
- MOD13A1 (500m_16_days_pixel_reliability)
- MOD13Q2 (250m_16_days_pixel_reliability)
- MOD14A1 (FireMask)
- MOD14A2 (FireMask)

- MYD10A1 (NDSI_Snow_Cover_Basic_QA)
- MYD10A2(Maximum_Snow_Extent)(Eight_Day_Snow_Cover)
- MYD13A1 (500m_16_days_pixel_reliability)
- MYD13Q2 (250m_16_days_pixel_reliability)
- MYD14A1 (FireMask)
- MYD14A2 (FireMask)

**Comment by gadomski** [comment]
> Do we want to include fillvalues (nodata) at all? Maybe within in the "classification.json" or a dedicated .json file? Or just skip it for now?

I'd say leave it off for now since we don't know exactly where it will go, and we can always come back and add it later.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: src/stactools/modis/fragments/MOD10A2/061/file-info.json
Comment (pholleway) [reply src/stactools/modis/fragments/MOD10A2/061/file-info.json]: That makes sense. Sounds good. I've removed the Eight Day Snow Cover from the "file-info.json" MOD10A2 and MYD10A2.

**Review by gadomski — APPROVED** [review]

**Pull request merged**
